File: src/main/java/com/example/sneakerx/entities/customClasses/AddressSnapshot.java
package com.example.sneakerx.entities.customClasses;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AddressSnapshot {
    private String recipientName;
    private String phone;
    private String provinceOrCity;
    private String district;
    private String ward;
    private String addressLine;
}

File: lib/src/models/custom/address_snapshot.dart
class AddressSnapshot {
  final String recipientName;
  final String phone;
  final String provinceOrCity;
  final String district;
  final String ward;
  final String addressLine;

  AddressSnapshot({
    required this.recipientName,
    required this.phone,
    required this.provinceOrCity,
    required this.district,
    required this.ward,
    required this.addressLine,
  });

  factory AddressSnapshot.fromJson(Map<String, dynamic> json) {
    return AddressSnapshot(
      recipientName: json['recipientName'],
      phone: json['phone'],
      provinceOrCity: json['provinceOrCity'],
      district: json['district'],
      ward: json['ward'],
      addressLine: json['addressLine'],
    );
  }
}